lib/Models/GlobeOrMap.js

```diff
@@ -246,12 +246,14 @@ GlobeOrMap.prototype._highlightFeature = function(feature) {
                   'marker-color': this.terria.baseMapContrastColor
                 };
 
+                var that = this;
                 var removeCallback = this._removeHighlightCallback = function() {
-                    catalogItem._hide();
-                    catalogItem._disable();
+                    that._highlightPromise.then(function() {
+                        catalogItem._hide();
+                        catalogItem._disable();
```

Comment:
I think this is still not quite right.  The problem is that `catalogItem._enable()` only gets called if `removeCallback === that._removeHighlightCallback`, which won't be true if a new feature has been selected in the meantime.  So even after waiting for the highlight promise to resolve, there's no guarantee that `_enable` and `_show` have been called, so an exception may still be raised.

However, now that the exception will be thrown inside a `then` callback, it will be silently ignored rather than causing feature picking to abort.  So that's definitely an improvement.

I think the only way to avoid the exception entirely is to track whether the layer has actually been enabled.  It's a bit ugly, but that can be done by slamming a property on the highlightPromise that gets set to true when the layer is enabled.  Maybe you can think of a cleaner way?